R/binary_classification.R

```diff
@@ -138,3 +138,210 @@ fbeta_score <- function(actual, predicted, beta = 1) {
     rec <- recall(actual, predicted)
     return((1 + beta^2) * prec * rec / ((beta^2 * prec) + rec))
 }
+
+#' Binary confusion matrix
+#'
+#' \code{cmat} Calculates a binary classification confusion matrix,
+#' comparing the predicted with the actual values for the classes.
+#' Assumes that 1 is used for the positive class and 0 for the
+#' negative class.
+#' Returns a \code{data.frame} with columns corresponding to the
+#' number of True Positives (\code{tp}), False Positives (\code{fp}),
+#' True Negatives (\code{tn}), and False Negatives (\code{fn})
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{sensitivity}} \code{\link{specificity}}
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' cmat(actual, predicted)
+
+cmat <- function(actual, predicted) {
+    tbl <- as.data.frame(table(actual, predicted))
+    cm <- data.frame(tp = NA, fn = NA, fp = NA, tn = NA)
+    cm[, c("tn", "fp", "fn", "tp")] <- tbl$Freq
+    cm
```

Comment:
Have you tested what happens with this function on a variety of different types of inputs? For example, when I set `actual <- c(1, 1, 1)` and `predicted <- c(1, 1, 1)`, I get a confusion matrix that does not look correct.

Do you want to try to fix the function, so that it is robust to different types of outputs?

Reply:
Implemented a bit more checking in the binary confusion_matrix function, should now complain if something different from 0s or 1s are fed to it as vectors